src/DepartmentService.Data/DepartmentUserRepository.cs

```diff
@@ -191,18 +212,27 @@ public Task<List<DbDepartmentUser>> GetAsync(List<Guid> usersIds, bool includeDe
 
     public Task<List<DbDepartmentUser>> GetAsync(IGetDepartmentsUsersRequest request)
     {
-      IQueryable<DbDepartmentUser> dbDepartmentUser = request.ByEntryDate.HasValue 
+      IQueryable<DbDepartmentUser> departmentUsersQuery = request.ByEntryDate.HasValue 
         ? _provider.DepartmentsUsers
             .TemporalBetween(
               request.ByEntryDate.Value,
               request.ByEntryDate.Value.AddMonths(1))
-            .Where(u => u.IsActive).Distinct()
-            .AsQueryable()
-        : _provider.DepartmentsUsers.AsQueryable().Where(du => du.IsActive);
+        : _provider.DepartmentsUsers.AsQueryable();
 
-      dbDepartmentUser = dbDepartmentUser.Where(du => request.DepartmentsIds.Contains(du.DepartmentId));
+      departmentUsersQuery = departmentUsersQuery.Where(du => request.DepartmentsIds.Contains(du.DepartmentId));
+
+      if (request.IncludePendingUsers)
+      {
+        departmentUsersQuery = from users in departmentUsersQuery.Where(du => du.IsActive || du.IsPending)
+                               group users by users.UserId into userGroup
+                               select userGroup.AsQueryable().OrderByDescending(u => u.IsActive).First();

```

Comment:
Isn't it queryable?

Reply:
no, it is IGrouping, and ef can't translate it without AsQueryable()

Reply:
I pulled your branch and remove AsQueryable method and it's OK

Reply:
tried again and it works, will remove it